src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -0,0 +1,47 @@
+import { E8S_PER_ICP } from '$lib/constants/icp.constants';
+import { type Token } from '@dfinity/utils';
+
+/**
+ * Truncates the given amount to 8 decimals.
+ *
+ * This is used to then convert the amount to a string or to a number afterwards.
+ */
+const ulpsToE8s = ({ ulps, decimals }: { ulps: bigint; decimals: number }): bigint => {
+	if (decimals === 8) {
+		return ulps;
+	} else if (decimals < 8) {
+		return ulps * 10n ** BigInt(8 - decimals);
+	}
+	return ulps / 10n ** BigInt(decimals - 8);
+};
+
+/**
+ * Calculates the maximum amount for a transaction.
+ *
+ * @param {Object} params
+ * @param {bigint | undefined} params.balance The balance of the account in E8S.
+ * @param {bigint | undefined} params.fee The fee of the transaction in E8S.
+ * @param {bigint | undefined}params.maxAmount The maximum amount of the transaction not counting the fees.
+ * @returns {number} The maximum amount for the transaction.
+ */
+export const getMaxTransactionAmount = ({
```

Comment:
I except Ethereum and ICP max amount calculation to be different, notably because on ICP calculing max amount for ckBTC and ckETH might involve additional fees. Are you sure we can use a single method for both networks?

Reply:
This function is just an helper to calculate the "final" amount once you have the balance and the fee. The fee calculation is done at the parent level of whatever calls this function.